black.py

```diff
@@ -1958,6 +1959,12 @@ def normalize_prefix(leaf: Leaf, *, inside_brackets: bool) -> None:
     leaf.prefix = ""
 
 
+def _sub_twice(regex: Pattern[str], replacement: str, string: str) -> str:
```

Comment:
1. Utility functions belong at the bottom of the file.
2. Add a reference to this function in docs/reference/functions.rst.
3. Note that I don't use any underscored names for functions in black.py. It's an application so the concept of private/public is not necessary. `sub_twice` reads a tiny bit nicer.
4. `string` would be more obvious if named `original`. We see it's a string from the type hint right next to it.
5. In the docstring, add a single leading line explaining what it does. I get that this is spurious for a single line function but it's going to be visible in the docs. Something simple like "Replace instances of `regex` in `original` with `replacement`, twice."